src/psp/report.rs

```diff
@@ -0,0 +1,141 @@
+/* SPDX-License-Identifier: MIT */
+/*
+ * Copyright (C) 2023 IBM
+ *
+ * Authors:
+ *   Claudio Carvalho <cclaudio@linux.ibm.com>
+ */
+
```

Comment:
This should only be done when requested. So it should be added to the svsm_request.rs file based on the attestation protocol value.

Reply:
@tlendacky : I started working on gluing the code here to the SVSM_ATTEST_SINGLE_SERVICE command etc. This will be extra commits on top of Claudio's work. Would you like to see all of it in the same PR? This is getting a bit big.

Reply:
We should probably break the pull request into two pull requests, one that creates the libcrt/openssl integration and one that supports the attestation protocol.

With multiple protocols, svsm_request.rs should probably be split into multiple files: svsm_request.rs and protocols/core.rs. Then the attestation protocol can be added as protocol/attestation.rs, etc, with everything called from svsm_request.rs.

Reply:
> With multiple protocols, svsm_request.rs should probably be split into multiple files: svsm_request.rs and protocols/core.rs.

I tried this extraction in #38 .